src/crystal/datum.cr

```diff
@@ -53,15 +53,30 @@ module Crystal
     # Raises if the underlying value is not an `Array` or `Hash`.
     def size : Int
       case object = @raw
-      when Array
-        object.size
-      when Hash
+      when Array, Hash
         object.size
       else
         raise "Expected Array or Hash for #size, not #{object.class}"
       end
     end
 
+    # Returns `true` if `self` does not currently have a value, or the value is `nil`.
+    #
+    # If the underlying value is an `Array` or `Hash` then `Enumerable#empty?` is invoked.
+    #
+    # If the underlying value is a `String`, then `String#empty?` is invoked.
+    #
+    # Otherwise, `false` is returned.
+    def empty? : Bool
```

Comment:
Something that bothers me of `JSON::Any` and `YAML::Any` is that there is no easy way to know if the value is nested. I think that that is needed.

I found the `empty?` a bit ambiguous. Because it delegates the empty to the raw value. It is not separating fully where `size` method will work or not.

The goal was to know if the `size` will raise or not, right?

I would prefer to add a `nested?` that might return the raw value as `Array | Hash | Nil` directly. So it's like equivalent to `as_nested? = as_h? || as_a?`.

That way we can use it either as a truthy check or with `case` in a comfortable manner.